src/main/java/appeng/me/storage/ItemCellInventory.java

```diff
@@ -0,0 +1,295 @@
+/*
+ * This file is part of Applied Energistics 2.
+ * Copyright (c) 2013 - 2014, AlgorithmX2, All rights reserved.
+ *
+ * Applied Energistics 2 is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Applied Energistics 2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Applied Energistics 2.  If not, see <http://www.gnu.org/licenses/lgpl>.
+ */
+
+package appeng.me.storage;
+
+
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+
+import appeng.api.AEApi;
+import appeng.api.config.Actionable;
+import appeng.api.exceptions.AppEngException;
+import appeng.api.implementations.items.IStorageCell;
+import appeng.api.networking.security.IActionSource;
+import appeng.api.storage.IMEInventory;
+import appeng.api.storage.IMEInventoryHandler;
+import appeng.api.storage.ISaveProvider;
+import appeng.api.storage.IStorageChannel;
+import appeng.api.storage.channels.IItemStorageChannel;
+import appeng.api.storage.data.IAEItemStack;
+import appeng.core.AEConfig;
+import appeng.core.AELog;
+
+
+public class ItemCellInventory extends AbstractCellInventory<IAEItemStack>
+{
+	protected ItemCellInventory( final NBTTagCompound data, final ISaveProvider container )
+	{
+		super( data, container, 8 );
+	}
+
+	private ItemCellInventory( final ItemStack o, final ISaveProvider container ) throws AppEngException
+	{
+		super( o, container, 8 );
+	}
+
+	public static IMEInventoryHandler getCell( final ItemStack o, final ISaveProvider container2 )
+	{
+		try
+		{
+			return new ItemCellInventoryHandler( new ItemCellInventory( o, container2 ) );
+		}
+		catch( final AppEngException e )
+		{
+			return null;
+		}
+	}
+
+	private static boolean isStorageCell( final ItemStack i )
+	{
+		if( i == null )
+		{
+			return false;
+		}
+
+		try
+		{
+			final Item type = i.getItem();
+			if( type instanceof IStorageCell )
+			{
+				return !( (IStorageCell) type ).storableInStorageCell();
+			}
+		}
+		catch( final Throwable err )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1166)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1181)